polaris-service/src/main/java/io/polaris/service/admin/PolarisAdminService.java

```diff
@@ -1739,7 +1739,7 @@ private boolean revokePrivilegeOnTableLikeFromRole(
       TableIdentifier identifier,
       PolarisEntitySubType subType,
       PolarisPrivilege privilege) {
-    PolarisEntity catalogEntity =
+    CatalogEntity ignored =
```

Comment:
I think we can remove these `ignored` variables, and replace it with something like this
```
    Preconditions.checkArgument(findCatalogByName(catalogName).isPresent(), "Catalog %s not found", catalogName);
```

Reply:
That would indeed work, but would raise IllegalArgumentException and the code currently throws NotFoundException. If we want such change, i would prefer that it's done in a  PR separately from pure code cleanup.

I wish Optional had ifEmpty method (just like it has ifPresent). the closest is `ifPresentOrElse`:

```java
findCatalogByName(catalogName)
    .ifPresentOrElse(
        ignored -> {},
        () -> {
          throw new NotFoundException("Parent catalog %s not found", catalogName);
        });
```

it happens that non-functional style is actually shorter

```java
al<CatalogEntity> catalog = findCatalogByName(catalogName);
if (catalog.isEmpty()) {
  throw new NotFoundException("Parent catalog %s not found", catalogName);
}
```

which one would you prefer?

Reply:
+1 to fix it in a follow-up PR. I like the non-functional one more.

Reply:
if we go with non-functional, i can fix it in this PR, since it's not a semantical code change.

Reply:
done!